asdf/commands/tests/test_remove_hdu.py

```diff
@@ -1,10 +1,17 @@
 import os
+import sys
 
 import numpy as np
+import pytest
 from astropy.io import fits
 
 from asdf.commands import remove_hdu
-from asdf.fits_embed import AsdfInFits
+from asdf.exceptions import AsdfDeprecationWarning
+
+with pytest.warns(AsdfDeprecationWarning, match="AsdfInFits has been deprecated.*"):
+    if "asdf.fits_embed" in sys.modules:
+        del sys.modules["asdf.fits_embed"]
+    from asdf.fits_embed import AsdfInFits
```

Comment:
Given all the discussion about `import asdf.fits_embed` shouldn't this be:
```suggestion
with pytest.warns(AsdfDeprecationWarning, match="AsdfInFits has been deprecated.*"):
    if "asdf.fits_embed" in sys.modules:
        del sys.modules["asdf.fits_embed"]
    import asdf.fitest_embed
    from asdf.fits_embed import AsdfInFits
```

Reply:
Similar to https://github.com/asdf-format/asdf/pull/1337#discussion_r1110074571 I changed the imports in this test to avoid the confusion.